src/components/secret-display/SecretDisplaySection.tsx

```diff
@@ -83,121 +97,86 @@ export default function SecretDisplaySection({
-                      weight="bold"
-                    >
-                      {lang.t('back_up.secret.copy_to_clipboard')}
-                    </Text>
-                  </Inline>
-                </CopyFloatingEmojis>
-              </Box>
-              <Stack alignHorizontal="center" space="19px (Deprecated)">
-                {/* @ts-ignore */}
-                <SecretDisplayCard seed={seed} type={type} />
+    case SecretDisplayStates.securedWithBiometrics:
+      return (
+        <SecretDisplayError
+          message={lang.t('back_up.secret.biometrically_secured')}
+        />
+      );
+    case SecretDisplayStates.revealed:
+      return (
+        <Wrapper>
+          <Box paddingBottom="19px (Deprecated)">
+            {/* @ts-expect-error JS component*/}
+            <CopyFloatingEmojis textToCopy={seed}>
+              <Inline alignVertical="center" space="6px">
+                <Icon
+                  name="copy"
+                  color={colors.appleBlue}
+                  style={styles.copyIcon}
+                />
                 <Text
-                  containsEmoji
-                  color="primary (Deprecated)"
+                  color="action (Deprecated)"
                   size="16px / 22px (Deprecated)"
                   weight="bold"
                 >
-                  👆{lang.t('back_up.secret.for_your_eyes_only')} 👆
-                </Text>
-                <Text
-                  align="center"
-                  color="secondary60 (Deprecated)"
-                  size="16px / 22px (Deprecated)"
-                  weight="semibold"
-                >
-                  {lang.t('back_up.secret.anyone_who_has_these')}
+                  {lang.t('back_up.secret.copy_to_clipboard')}
                 </Text>
-              </Stack>
-            </>
-          ) : (
-            <LoadingSpinner color={colors.blueGreyDark50} />
-          )}
-        </Box>
-      ) : (
-        renderStepNoSeeds()
-      )}
-    </>
-  );
+              </Inline>
+            </CopyFloatingEmojis>
+          </Box>
+          <Stack alignHorizontal="center" space="19px (Deprecated)">
+            <SecretDisplayCard seed={seed ?? ''} type={walletType} />
+            <Text
+              containsEmoji
+              color="primary (Deprecated)"
+              size="16px / 22px (Deprecated)"
+              weight="bold"
+            >
+              👆{lang.t('back_up.secret.for_your_eyes_only')} 👆
+            </Text>
+            <Text
+              align="center"
+              color="secondary60 (Deprecated)"
+              size="16px / 22px (Deprecated)"
+              weight="semibold"
+            >
+              {lang.t('back_up.secret.anyone_who_has_these')}
+            </Text>
+          </Stack>
+        </Wrapper>
+      );
+    default:
+      logger.error(
+        new RainbowError(
+          'Secret display section tires to present an unknown state'
+        )
+      );
```

Comment:
Not sure what you think about it, is there a better way to satisfy TypeScript and also make sure we are safe?

Reply:
Seems OK to me. If this ever happens we def want to know, but we don't necessarily need to blow up the UI, so returning `null` seems good to me. Any other concerns?

```suggestion
      logger.error(
        new RainbowError(
          'Secret display section tries to present an unknown state'
        )
      );
```